### PR TITLE
HWweek  6 and TEBOW IT

### PR DESCRIPTION
**Pull request by Lizzy95**

### CODE REVIEW

File: HWweek6.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+class TreeNode(object):
```
Comment (charientism) [inline HWweek6.py:1]: Your code is very clean/concise :)

That makes this a bit of a nitpick, but don't forget to leave comments in your pull requests. It helps me see your thought-process more clearly, and therefore works as a good way to practice the "Talk" step.

File: HWweek6.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+def isSubtree(s, t):
+    if s is None:
+        return False
+    if t is None:
+        return False
+    if (isequal(s, t)):
+        return True
+    return isSubtree(s.left, t) or isSubtree(s.right, t)
+
+#Test 1
```
Comment (charientism) [inline HWweek6.py:23]: Thanks for uploading your written TEBOW IT notes! 

You may already be doing this, but make sure that you practice writing our your implementation via pen and paper/whiteboard before you code it up on a computer. It's closer to what you'll be doing for on-campus interviews, and let's you know if there's a difference in your writing/typing speeds (I type much faster than I write, for example, so writing out the answers gives me a more accurate picture of how long the problem would take me in an interview).

File: HWweek6.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+T = TreeNode(None)
+S = TreeNode(None)
+print isSubtree(S, T)
+
+#Test 6
+T = TreeNode(1)
+S = TreeNode(1)
+print isSubtree(S, T)
+
+#Test 7
```
Comment (charientism) [inline HWweek6.py:81]: Great test cases (keeping things small, watching out for edge cases, picking good positive/negative examples)! :)

File: HWweek6.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+T = TreeNode(None)
+S = TreeNode(1)
+print isSubtree(S, T)
+
+
+
```
Comment (charientism) [inline HWweek6.py:87]: Awesome job with this assignment! :D

**Review by charientism — APPROVED** [review]

File: HWweek6.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
+class TreeNode(object):
+    def __init__(self, x):
+        self.val = x
+        self.left = None
+        self.right = None
+
+def isequal(rootS, rootT):
+    if rootS is None and rootT is None:
```
Comment (maknoon) [inline HWweek6.py:8]: Quick optimization tip: because I actually made the same logic comparison here & Carla pointed out that we can actually only just check if `rootT` is`None`. Because if T is null then it will always be a subtree & always return true!

**Comment by maknoon** [comment]
Great job on the test cases! Very detailed. :) A nice-to-have is maybe putting some more comments on the equivalence class test case descriptions in the code to compare them side by side. I also like how your code is really clean & easy-to-follow (Python is <3)!
